usersession/client/client_test.go

```diff
@@ -448,12 +449,37 @@ func (s *clientSuite) TestServicesStopFailure(c *C) {
 }
 
 func (s *clientSuite) TestPendingRefreshNotification(c *C) {
+	n := 0
 	s.handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		n++
 		c.Assert(r.URL.Path, Equals, "/v1/notifications/pending-refresh")
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(200)
 		w.Write([]byte(`{"type": "sync"}`))
 	})
 	err := s.cli.PendingRefreshNotification(context.Background(), &client.PendingSnapRefreshInfo{})
 	c.Assert(err, IsNil)
+	// two calls because clientSuite simulates two user sessions (two
+	// snapd-session-agent.socket sockets).
+	c.Check(n, Equals, 2)
+}
+
+func (s *clientSuite) TestFinishRefreshNotification(c *C) {
+	n := 0
+	s.handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		n++
```

Comment:
Similar to https://github.com/snapcore/snapd/pull/11072#discussion_r755807596, this increment looks potentially racy. It should be protected by a mutex or use an atomic increment.

Reply:
Fixed.